File: solutions/problem1.py
# Two Sum — LeetCode Problem 1
# Default fallback solution used by Streak Guardian
# Replace this with any problem you prefer

class Solution:
    def twoSum(self, nums, target):
        seen = {}
        for i, n in enumerate(nums):
            if target - n in seen:
                return [seen[target - n], i]
            seen[n] = i

File: solutions/problem1.cpp
// Two Sum — LeetCode Problem 1 (C++)
// Default fallback solution used by Streak Guardian
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> seen;
        for (int i = 0; i < (int)nums.size(); i++) {
            if (seen.count(target - nums[i]))
                return {seen[target - nums[i]], i};
            seen[nums[i]] = i;
        }
        return {};
    }
};
